client/src/containers/home/scroll/index.tsx

```diff
@@ -23,7 +24,7 @@ const ScrollItem = ({ children, step, onChange }: ScrollItemProps) => {
   }, [s, step, inView, onChange]);
 
   return (
-    <section ref={ref} className="min-h-[100vh]">
+    <section ref={ref} id={id} className="min-h-[100vh]">
```

Comment:
Instead of passing the id from the outside. Don't you think it will be easier to generate it based on the step number?